src/common/hist_util.cu

```diff
@@ -113,9 +127,13 @@ struct GPUSketcher {
 
    public:
     DeviceShard(int device, bst_uint row_begin, bst_uint row_end,
-                tree::TrainParam param) :
+                tree::TrainParam param, SketchContainer *sketch_container) :
       device_(device), row_begin_(row_begin), row_end_(row_end),
-      n_rows_(row_end - row_begin), param_(std::move(param)) {
+      n_rows_(row_end - row_begin), param_(std::move(param)), sketch_container_(sketch_container) {
+    }
+
+    inline size_t GetRowStride() {
```

Comment:
Any reason to have a method here? If the purpose is to make `row_stride_` accessible, consider making `DeviceShard` a struct.

Reply:
what is the issue with having this accessor? i can consider making it a const member function, if any, as the state isn't getting mutated. deviceshard already delineates what is public and private and i do not want to loosen the accessibility restrictions without fully understanding why it was made a class in the first place.